disperser/apiserver/server.go

```diff
@@ -315,84 +311,103 @@ func (s *DispersalServer) getAccountRate(origin, authenticatedAddress string, qu
 
 }
 
-func (s *DispersalServer) checkRateLimitsAndAddRates(ctx context.Context, blob *core.Blob, origin, authenticatedAddress string) error {
+type limiterInfo struct {
+	Name     string
+	QuorumID core.QuorumID
+}
 
-	// TODO(robert): Remove these locks once we have resolved ratelimiting approach
-	s.mu.Lock()
-	defer s.mu.Unlock()
+func (s *DispersalServer) checkRateLimitsAndAddRatesToHeader(ctx context.Context, blob *core.Blob, origin, authenticatedAddress string) (bool, string, error) {
```

Comment:
nit: it would also be helpful to document what's being returned here, and also noting that this method assumes that the `blob` has already been validated

Reply:
Added